<launch>
    
    <include
    file="$(find gazebo_ros)/launch/empty_world.launch" />
    
    <node
    name="world_to_base_link"
    pkg="tf"
    type="static_transform_publisher"
    args="0 0 0 0 0 0 world base_link 40" />
  
    <node
    name="tf_footprint_base"
    pkg="tf"
    type="static_transform_publisher"
    args="0 0 0 0 0 0 base_link base_footprint 40" />
    <!-- Spawn the robot in gazebo -->
    <node
    name="spawn_model"
    pkg="gazebo_ros"
    type="spawn_model"
    args="-file $(find movit_full_sampling)/urdf/full_sampling_urdf.urdf -urdf -model full_sampling_urdf"
    output="screen" />
   

  <arg
    name="model" />
  <param
    name="robot_description"
    textfile="$(find movit_full_sampling)/urdf/full_sampling_urdf.urdf" />
  
  <node
    name="robot_state_publisher"
    pkg="robot_state_publisher"
    type="robot_state_publisher" />
  <node
    name="rviz"
    pkg="rviz"
    type="rviz"
    args="-d $(find movit_full_sampling)/config/urdf.rviz" />

    <include file = "$(find movit_full_sampling)/launch/move_group.launch" />
    
    <!-- Load joint controller configurations from YAML file to parameter server -->
    <rosparam file="$(find movit_full_sampling)/config/teleop_controllers.yaml" command="load" ns="/"/>

    <!-- Load the controllers with namespace -->
    <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen" 
    args="joint_state_controller joint_1_position_controller joint_2_position_controller joint_3_position_controller joint_4_position_controller joint_5_position_controller joint_ree_position_controller joint_lee_position_controller --timeout 500" />

    <!-- Start the joint state publisher -->
    <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" />

    <node pkg="movit_full_sampling" type="joint_translator.py" name="joint_command_splitter" output="screen"/>
    <node pkg="movit_full_sampling" type="IKSolver.py" name="ik_controller_node" output="screen"/>
</launch>